src/main/java/com/splunk/hecclient/Indexer.java

```diff
@@ -156,16 +159,13 @@ public boolean send(final EventBatch batch) {
     public synchronized String executeHttpRequest(final HttpUriRequest req) {
         CloseableHttpResponse resp;
         if (hecConfig.kerberosAuthEnabled()) {
-            if (config == null) {
-                defineKerberosConfigs();
-            }
-            Set<Principal> princ = new HashSet<Principal>(1);
-            princ.add(new KerberosPrincipal(hecConfig.kerberosUser()));
-            Subject sub = new Subject(false, princ, new HashSet<Object>(), new HashSet<Object>());
             try {
-                LoginContext lc = new LoginContext("", sub, null, config);
-                lc.login();
-                Subject serviceSubject = lc.getSubject();
+                if (config == null) {
+                    defineKerberosConfigs();
+                }
+                if(serviceSubject == null){
```

Comment:
Instead of doing this here by checking for evvery request, you can initialize this in constructor if necessary.

Reply:
Correct, see if it's possible to move the kerb config definition and the subject initialization in constructor, if not, keeping it here with a null check also makes sense.